app/celery/tasks.py

```diff
@@ -287,58 +297,57 @@ def save_smss(self, service_id: Optional[str], signed_notifications: List[Any],
 
     check_service_over_daily_message_limit(KEY_TYPE_NORMAL, service)
 
-    research_mode = service.research_mode  # type: ignore
-
     current_app.logger.info(f"Sending following sms notifications to AWS: {notification_id_queue.keys()}")
-    for notification in saved_notifications:
-        queue = notification_id_queue.get(notification.id) or template.queue_to_use()  # type: ignore
+    for notification_obj in saved_notifications:
+        check_if_request_would_put_service_over_daily_sms_limit(KEY_TYPE_NORMAL, service, notification_obj.billable_units)  # type: ignore
```

Comment:
Here is the only "real" change I made to the logic, to add the additional limit check